### ISSUE TITLE
After I harvest, ask if I have seeds to trade

### ISSUE THREAD

**Issue by CloCkWeRX** [Growstuff/growstuff#862]
#501 related.

If I harvest a few kg of pumpkin, it would be good to be given a howto on harvesting the seeds etc (a simple google link to avoid us having to capture the data).

Some time later, it would be good to be prompted to see if I had seeds to trade (in the email what have you planted reminder)

**Comment by maco** [comment]
This probably only makes sense if the harvested part is "fruit" (wait, do we have a "seed head" option? because alliums do put out seed heads hmmm) but not if it's, say, leaves. Clipping my mint doesn't mean it went to seed, and not everything *does* go to seed.

**Comment by pozorvlak** [comment]
I suspect any attempt to limit this to "fruit" etc will be defeated by the
infinite cussedness of biology. Maybe we need a `harvesting_produces_seeds`
boolean field on `Crop`?

**Comment by maco** [comment]
Then there's the stuff that produces seeds when you fail to harvest it,
because failing to harvest made it bolt... bleh

**Comment by andru** [comment]
There's also the problem that most vegetable crops cross freely, making harvest a bit late to offer seed saving advice. 

It's really only Tomatoes and a select few others which self pollinate almost exclusively, making seed saving a convenient afterthought. For the rest, providing seed saving advice after someone has planted 3 kinds of _Cucurbita pepo_ (e.g. A zuccini, a crookneck and an acorn squash) or Jalapeños alongside their Bell Pepper Capsicums is going to leave someone with a seed bank of screwy genetics which really shouldn't be shared.

I think the idea of encouraging seed saving is _great_ though... maybe the advice could be offered at the trigger most applicable to the crop in question?  E.g. for closed self pollinating flowers like Tomatoes it's advice on saving seed given at harvest; for insect pollinated flowers like squashes it's advice given on hand pollinating flowers given when they are first flowering; for wind pollinated or insect pollinated flowers which are difficult to hand pollinate like corn or brassicas, it's advice on techniques for isolating the crop and the distance required between plantings of different varieties required to get a good harvest of true seed.

That makes it a lot more complicated, but I'm not sure how useful of a feature it would be otherwise?

**Comment by andru** [comment]
Or, for something much less complicated, maybe each crop has some seed saving instructions - detailing what kind of pollinator the plant is, the work involved in keeping the seed true to type, and some details on harvesting and cleaning the seed - and that information is made prominent when first planting the crop so the user is exposed to that information early and can plan for it accordingly?

**Comment by pozorvlak** [comment]
"Each crop has some seed saving instructions" sounds like the simplest
thing that could possibly work - let's start with that.

**Comment by CloCkWeRX** [comment]
So I was thinking the data collection would be tedious; and we might be best off sticking a link like
```
<a href="https://www.google.com/search?q=harvest+zucchini+seeds" target="_blank" />
```
on the assorted crop pages

**Comment by andru** [comment]
Maybe that could be a fall-back? Or maybe we show nothing at all unless a crop has some seed saving guidance defined. 

There are a lot of crops, for example, which you can't or rarely propogate by seed (peppermint hybrids, most fruit bushes/trees, etc), so a link to google for seed saving advice could be misleading.

Lots of seed saving instructions are fairly generic and apply broadly to a whole genera or sub group, so it's easy to cover a large number of common garden vegetables with little work.

Does the crop model have any concept of hierarchy? i.e. can we set some property on the genus _Cucurbita_ and have that property propagate to all species within that genus? If so, that would be an easy way to make available some details to a large number of crops very easily.

**Comment by maco** [comment]
Yes, there are parent & child crops. How well-structured they are depends
on how much extra work the crop wrangler who approved a particular crop
felt like doing.

**Comment by pozorvlak** [comment]
Also, the crop hierarchy doesn't always line up with genus/species
boundaries (deliberately - consider all the different varieties of
*Brassica oleracea*!)

**Comment by andru** [comment]
> Also, the crop hierarchy doesn't always line up with genus/species deliberately - consider all the different varieties of *Brassica oleracea*

Going a bit off topic here for a moment, so please excuse the diversion but just want to point out that the number of varieties, even morphologically very distinct varieties, within a species is not a barrier to a concept of hierarchy in a crop model.

In this case ICNCP [cultivar Groups](https://en.wikipedia.org/wiki/Cultivar_group) are very helpful in organising a hierarchy for the many Brassica varieties, grouping by shared traits (Broccolis, Cabbages, Kales, and so on)...

e.g. _B. oleracea_ varieties cultivated primarily for the flowering heads fall non-exclusively within the _Italica_ cultivar group, the hierarchy for 'Early Purple Sprouting' Broccoli is...

Brassica > _B. oleracea_ > _B. oleracea_ Italica Group > _B. oleracea_ 'Early Purple Sprouting'

**Comment by pozorvlak** [comment]
>
> the number of varieties, even morphologically very distinct varieties,
> within a species is not a barrier to a concept of hierarchy in a crop model.
>
Indeed not, and we already have a concept of hierarchy in our crop model
:-) The thing we wanted to avoid was a naïve "crop == species" constraint,
or something similar.

> In this case ICNCP cultivar Groups
> <https://en.wikipedia.org/wiki/Cultivar_group> are very helpful in
> organising a hierarchy for the many Brassica varieties, grouping by shared
> traits (Broccolis, Cabbages, Kales, and so on)...
>
Oh, excellent - I didn't know about those, and they look very useful!
Perhaps in future we should restructure our crop hierarchy along more
ICNCP-like lines.

**Comment by andru** [comment]
> Indeed not, and we already have a concept of hierarchy in our crop model
:-) The thing we wanted to avoid was a naïve "crop == species" constraint,
or something similar.

👍 🙌

**Comment by Br3nda** [comment]
I do want to track the lineage of my food (and this culturally important to many peoples in the world). So linking a "seed" to a planting seems the first step. Along with prompts to enter it at harvest time. 

There lots of places the ui can prompt more generally, creating frictionless edits, eg tomato plantings that are more than a year old should nudge the user (in browser) to mark as finished.

**Comment by Br3nda** [comment]
Seed -> planting ->seed -> planting-> seed... 
All many to many.
With harvests along the way.

**Comment by Br3nda** [comment]
Entering past data would be good too. Recording that this kamokamo came from grandad's seed saving...

**Comment by Br3nda** [comment]
I wonder if we can have the seed saving advice hosted on openfarm.cc, and either link to it, or request from their api.

**Comment by CloCkWeRX** [comment]
https://www.growstuff.org/harvests/clockwerx-purslane

Kind of got solved - show harvest page features save seeds as an option by virtue of the crop partial.

Original issue, solved.

The guide for saving the seeds... Might add a generic field to crops
